### ISSUE TITLE
customizable LinkPager

### ISSUE THREAD

**Issue by nkovacs** [yiisoft/yii2#9073]
The current output is `1 << 3 4 5 6 7 8 >> 10`. I'd like to change it to `<< 1 ...  3 4 5 6 7 8 ... 10 >>`, but I have to copy and overwrite all of `renderPageButtons`, which is a lot of duplicated code.

It would be great if I could customize LinkPager with a template like this:

`"{prevPage}{firstPage}{firstEllipsis}{pageRange}{lastEllipsis}{lastPage}{nextPage}"`

**Comment by nkovacs** [comment]
Actually, it currently outputs `1 << 1 2 3 4 5 >> 5 `, that is if `firstPageLabel` is true, it might output the first page twice, once as the first page link, and once in the inner range. So there should be an option to exclude the first and last page from the inner range.

**Comment by YaroslavShapoval** [comment]
Can we pass settings to LinkPager when creating ActiveDataProvider in action?

**Comment by cebe** [comment]
LinkPager is the widget configured in [GridView::$pager](http://www.yiiframework.com/doc-2.0/yii-widgets-baselistview.html#$pager-detail) so this is, where you pass the settings.

**Comment by YaroslavShapoval** [comment]
Thanks.

What about the issue. I do not think that templates of pagination should be implemented, as there is no so often need to change the order of the elements in pagination.

Maybe we need to add boolean parameter for display dots (only if first and last page labels are enabled), like
```php
# Show dots if the first or last elements are not currently visible
$displayDots = false;
```

It may look like this (for $maxButtonCount = 7):
```
<<  1 2 3 4 5 ... 10 >>
<<  1 ... 5 6 7 ... 10 >>
<<  1 ... 6 7 8 9 10 >>
```

Or maybe it should be another parameter with the same behavior:
```php
# Display first and last page labels near internal pages
$extremeLabelsInsideRange = false;
```

And maybe we need to add boolean parameter for hide first and last page label if it is start or end of pagination, like
```php
# Hide first and last pages if their labels is enabled
$hideExtremeItems = false;
```

Or maybe it should be done without parameter, just hide first item if `$firstPageLabel` is enabled and hide last item if `$lastPageLabel` is enable.
It may look like this:
```
1 << 2 3 4 5 >> 6
```

**Comment by samdark** [comment]
Won't be implemented in 2.0.
